src/services/auth.service.js

```diff
@@ -79,3 +83,57 @@ function register(name, email, phone, username, password1, password2) {
             }
         });
 }
+
+function checkUser(userId) {  
+     return axios.get(
+        stringFormat(`${config.commonConfig.$apiUrl}/${config.userConfig.api.checkUser.endpoint}`, userId)
+    )
+        .then((response) =>{
+            console.log('response =', response.data);
+            if ('username' in response.data) {
+                return response.data.username;
+            } else {
+                throw config.messagingConfig.messages.error.unknown_error;
+            }
+        })
+        .catch(() => {
+            throw stringFormat(config.messagingConfig.messages.error.does_not_exist_error, 'user');
+        }) 
+}
+
+function sendOTP(username) {
+    return axios.post(
+        stringFormat(`${config.commonConfig.$apiUrl}/${config.userConfig.api.sendOtp.endpoint}`, username)
+   )  
+}
+
+function verifyOtp(username, otp){
+    return axios.put(
+        stringFormat(`${config.commonConfig.$apiUrl}/${config.userConfig.api.verifyOtp.endpoint}`, username, otp)
+    )
+    .then((response) => {
+        const data = response.data;
+        console.log(data);
+        return true;
+    })
+    .catch(() => {
+        throw config.messagingConfig.messages.error.otp_not_match;
+    }) 
+}
+
+function resetPassword(username, password){
+    return axios.put(
+        stringFormat(`${config.commonConfig.$apiUrl}/${config.userConfig.api.resetPassword.endpoint}`, username),
+        { password: password }
+    )
+        .then(response => {
+            const data = response.data;
+            if (response.data) {
+                localStorage.setItem('user', JSON.stringify(data));
```

Comment:
Are you sure that this API sends back user details in response ? Can you try out this API in postman and let us know.